libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookHelper.cs

```diff
@@ -75,7 +75,7 @@ public static Activity ProcessSingleMessage(FacebookMessage message)
                 return null;
             }
 
-            if (message.Sender == null)
+            if (message.Sender == null && message.Optin != null && message.Optin.UserRef != null)
```

Comment:
Maybe not, but shouldn't `message.Optin` be `message.Option` instead?

Reply:
Not in this case, is weird, but Facebook returns a variable with this name.